sbi/simulators/simutils.py

```diff
@@ -122,78 +518,9 @@ def simulate_in_batches(
-                x = torch.from_numpy(x)
-            if simulation_batch_size == 1:
-                # squeeze in case simulator provides an additional dimension for single parameters,
-                # e.g. in linearGaussian example. Then prepend a dimension to be able to concatenate.
-                # XXX: here we are squeezing and unsqueezing on the same dim, no?
-                x = torch.squeeze(x, dim=0).unsqueeze(0)
-        # collect batches in list
-        all_x.append(x)
-
-    return torch.tensor(parameters), torch.cat(all_x)
-
-
-# XXX is this a utility function for testing? -> specific utils module
-def get_simulator_prior_and_groundtruth(task):
-
-    if task == "nonlinear-gaussian":
-        simulator = simulators.NonlinearGaussianSimulator()
-        prior = BoxUniform(
-            low=-3 * torch.ones(simulator.parameter_dim),
-            high=3 * torch.ones(simulator.parameter_dim),
-        )
-        # ground truth parameters as specified in 'Sequential Neural Likelihood' paper.
-        ground_truth_parameters = torch.tensor([-0.7, -2.9, -1.0, -0.9, 0.6])
-        # ground truth observation using same seed as 'Sequential Neural Likelihood' paper.
-        ground_truth_observation = torch.tensor(
-            [
-                -0.97071232,
-                -2.94612244,
-                -0.44947218,
-                -3.42318484,
-                -0.13285634,
-                -3.36401699,
-                -0.85367595,
-                -2.42716377,
-            ]
-        )
-
-    elif task == "nonlinear-gaussian-gaussian":
-        simulator = simulators.NonlinearGaussianSimulator()
-        prior = MultivariateNormal(loc=torch.zeros(5), covariance_matrix=torch.eye(5))
-        # ground truth parameters as specified in 'Sequential Neural Likelihood' paper.
-        ground_truth_parameters = torch.tensor([-0.7, -2.9, -1.0, -0.9, 0.6])
-        # ground truth observation using same seed as 'Sequential Neural Likelihood' paper.
-        ground_truth_observation = torch.tensor(
-            [
-                -0.97071232,
-                -2.94612244,
-                -0.44947218,
-                -3.42318484,
-                -0.13285634,
-                -3.36401699,
-                -0.85367595,
-                -2.42716377,
-            ]
-        )
-
-    elif task == "linear-gaussian":
-        dim, std = 20, 0.5
-        simulator = lambda theta: linear_gaussian(theta, std=std)
-        prior = MultivariateNormal(
-            loc=torch.zeros(dim), covariance_matrix=torch.eye(dim)
-        )
-        ground_truth_parameters = torch.zeros(dim)
-        ground_truth_observation = simulator(ground_truth_parameters)
-    else:
-        raise ValueError(f"'{task}' simulator choice not understood.")
+            xs.append(simulator(batch))
 
-    return simulator, prior, ground_truth_parameters, ground_truth_observation
+    return torch.tensor(parameters), torch.cat(xs)
```

Comment:
So in the end we have here the copy and hence the warning?

Reply:
good point, I forgot the essential part here.